rllib/models/specs/specs_base.py

```diff
@@ -82,6 +83,33 @@ def full_shape(self) -> Tuple[int]:
         """Returns a `tuple` specifying the concrete tensor shape (only ints)."""
         return self._full_shape
 
+    def rdrop(self, n: int) -> "TensorSpec":
+        """Returns of copy of TensorSpec with the rightmost
+        n dimensions removed.
+
+        Args:
+            n: The number of dimensions to remove from the right
+
+        Raises:
+            IndexError: If n is greater than the number of indices in self
+        """
+        self = deepcopy(self)
+        self._expected_shape = self.shape[:-n]
```

Comment:
Nit: Can we add a check that n is positive?

Reply:
Otherwise, users may think they have to pass in a negative index to indicate "from the right".